Sources/TuistCache/Utilities/CacheXCFrameworkBuilder.swift

```diff
@@ -37,9 +41,10 @@ public final class CacheXCFrameworkBuilder: CacheArtifactBuilding {
         // Create temporary directories
         return try await FileHandler.shared.inTemporaryDirectory { temporaryDirectory in
 
-            // Build for the simulator
+            // Build for the simulator - if required
             var simulatorArchivePath: AbsolutePath?
-            if platform.hasSimulators {
+            let buildForSimulator = self.cacheOutputType == .xcframework || self.cacheOutputType == .simulatorXCFramework
```

Comment:
```suggestion
            let buildForSimulator = platform.hasSimulators && self.cacheOutputType.shouldBuildForSimulator
```
 let's add a computed variable on `CacheOutputType`, and use a switch there so we need to consider all cases